### ISSUE TITLE
cleanup after successful preview (binary) push

### ISSUE THREAD

**Issue by fun3md** [barasher/picdexer#17]
path "/tmp" inside docker container fills with resized preview files for binary upload

this works quite well
```golang
        if _, err := io.Copy(part, input); err != nil {
		return err
	}else{
		input.Close()
		if err := os.Remove(f); err != nil{
			return err
		}
	}
```

```golang
func (p pusher) push(f string, key string) error {
	body := new(bytes.Buffer)
	mpart := multipart.NewWriter(body)
	part, err := mpart.CreateFormFile("file", key)
	if err != nil {
		return err
	}

	input, err := os.Open(f)
	if err != nil {
		return err
	}
	defer input.Close()
	if _, err := io.Copy(part, input); err != nil {
		return err
	}else{
		input.Close()
		if err := os.Remove(f); err != nil{
			return err
		}
	}
	if err = mpart.Close(); err != nil {
		return err
	}

	req, err := http.NewRequest("POST", p.url, body)
	if err != nil {
		return err
	}
	req.URL.Path = fmt.Sprintf("/key/%s", key)
	req.Header.Set("Content-type", fmt.Sprintf("multipart/form-data; boundary=%s", mpart.Boundary()))

	resp, err := p.httpClient.Do(req)
	if err != nil {
		return err
	}
	defer resp.Body.Close()
	if resp.StatusCode != http.StatusNoContent {
		return fmt.Errorf("Unexpected http status (%v)", resp.StatusCode)
	}
	return nil
}
```

**Comment by barasher** [comment]
Resized files are now properly deleted in commit 1f31a67e4da1d00ca52976c6c9d4768c42daa429
Thanks for the feedback !
I'll close the issue when the version will be "officially" released :p

**Comment by fun3md** [comment]
thanks, will test with the new version
and i will delete my pull request, after seeing your commit, the changes and the embarrassment i feel about the crap i cobbled together. :D ... but hey, it worked ... ^^

**Comment by barasher** [comment]
Lol, don't feel embarassed, your implementation just had an issue when picture resizing is disabled: the source file would have been deleted.

**Comment by barasher** [comment]
Fixed in commit 1ab3c6b16cd3b76f681f27c504d626c65fe2bc0e
Will be released in 2.0.0

**Comment by barasher** [comment]
https://github.com/barasher/picdexer/tree/v2.0.0
